src/github.com/getlantern/filepersist/filepersist.go

```diff
@@ -0,0 +1,116 @@
+// package filepersist provdies a mechanism for persisting data to a file at a
+// permanent location
+package filepersist
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/getlantern/golog"
+)
+
+var (
+	log = golog.LoggerFor("filepersist")
+)
+
+// Save saves the given data to the file at filename. If an existing file at
+// that filename already exists, this simply chmods the existing file to match
+// the given fileMode and otherwise leaves it alone.
+func Save(filename string, data []byte, fileMode os.FileMode) error {
```

Comment:
This logic is common between byteexec and the new runhide packages, so I factored it out.